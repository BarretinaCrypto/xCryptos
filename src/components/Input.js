import { array, number, oneOfType } from 'prop-types';
import React from 'react';
import { TextInput } from 'react-native';
import styles from './Input.style';

const Input = ({ style, ...inherit }) => (
  <TextInput
    {...inherit}
    autoCorrect={false}
    autoCapitalize="none"
    keyboardType="numeric"
    underlineColorAndroid="transparent"
    placeholderTextColor={undefined}
    style={[
      styles.input,
      (!inherit.editable && styles.disabled),
      style,
    ]}
  />
);

Input.propTypes = {
  style: oneOfType([array, number]),
};

Input.defaultProps = {
  style: [],
};

export default Input;
