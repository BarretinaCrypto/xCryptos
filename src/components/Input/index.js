import { array, number, oneOfType } from 'prop-types';
import React from 'react';
import { TextInput } from 'react-native';

import styles from './style';

const Input = ({ style, ...inherit }) => (
  <TextInput
    keyboardType="numeric"
    autoCorrect={false}
    autoCapitalize="none"
    underlineColorAndroid="transparent"
    placeholderTextColor={undefined}
    {...inherit}
    style={[
      styles.input,
      (!inherit.editable && styles.disabled),
      style,
    ]}
  />
);

Input.propTypes = {
  style: oneOfType([array, number]),
};

Input.defaultProps = {
  style: [],
};

export default Input;
