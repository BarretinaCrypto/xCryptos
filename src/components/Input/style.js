import { StyleSheet } from 'react-native';

import { THEME } from '../../config';

const { UNIT } = THEME;

export default StyleSheet.create({
  input: {
    fontSize: THEME.FONT.SIZE.LARGE,
    paddingVertical: UNIT,
    paddingHorizontal: UNIT / 4,
    width: '100%',
  },

  disabled: {
    color: THEME.COLOR_SECONDARY,
  },
});
