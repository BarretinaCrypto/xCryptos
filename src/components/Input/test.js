import 'react-native';
import React from 'react';
import renderer from 'react-native-test-utils';

import Input from './index';

jest.mock('TouchableOpacity', () => 'TouchableOpacity');

it('renders by default', () => {
  const tree = renderer(<Input />).toJSON();
  expect(tree).toMatchSnapshot();
});
