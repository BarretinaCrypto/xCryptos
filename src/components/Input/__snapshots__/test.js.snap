// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders by default 1`] = `
<TextInput
  allowFontScaling={true}
  autoCapitalize="none"
  autoCorrect={false}
  keyboardType="numeric"
  placeholderTextColor={undefined}
  style={
    Array [
      Object {
        "fontSize": 20,
        "paddingHorizontal": 2.5,
        "paddingVertical": 10,
        "width": "100%",
      },
      Object {
        "color": "rgba(0, 0, 0, 0.5)",
      },
      Array [],
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
